node-core/services/registry/registry.go

```diff
@@ -85,6 +87,23 @@ func (s *Registry) StartAll(ctx context.Context) error {
 	return nil
 }
 
+func (s *Registry) StopAll() error {
+	s.logger.Info("Stopping services", "num", len(s.serviceTypes))
+	for _, typeName := range s.serviceTypes {
+		s.logger.Info("Stopping service", "type", typeName)
+		svc := s.services[typeName]
+		if svc == nil {
+			s.logger.Error("service not found", "type", typeName)
+			continue
+		}
+
+		if err := svc.Stop(); err != nil {
```

Comment:
@fridrik01 we are assuming that `StopAll` won't ever be called before `Start` here. Should we explicitly enforce this? With some kind of  `started` atomic bool

Reply:
I left a comment that it should be safe to call Stop on a service that was never started.